src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -171,6 +171,10 @@ func (r *ClusterReconciler) Reconcile(
 	}
 	pki := certmanager.NewPki(r.Client, &redpandaCluster, headlessSvc.HeadlessServiceFQDN(r.clusterDomain), clusterSvc.ServiceFQDN(r.clusterDomain), r.Scheme, log)
 	sa := resources.NewServiceAccount(r.Client, &redpandaCluster, r.Scheme, log)
+	configMapResource := resources.NewConfigMap(r.Client, &redpandaCluster, r.Scheme, headlessSvc.HeadlessServiceFQDN(r.clusterDomain), proxySuKey, schemaRegistrySuKey, log)
+	configMapHashGetter := func() (string, error) {
+		return configMapResource.GetConfigHash(ctx)
+	}
```

Comment:
Why Getter does not takes context as an input? Couldn't we pass configMapResource.GetConfigHash to statefulset resource?